src/main.lua

```diff
@@ -3217,4 +3246,178 @@ addEventingHandler("paginatedDelegations", utils.hasMatchingTag("Action", "Pagin
 	})
 end)
 
+--- PRIMARY NAMES
+addEventingHandler("removePrimaryName", utils.hasMatchingTag("Action", ActionMap.RemovePrimaryNames), function(msg)
+	local names = msg.Tags.Names and utils.splitAndTrimString(msg.Tags.Names, ",") or nil
+	local from = utils.formatAddress(msg.From)
+	-- TODO: names must be provided
```

Comment:
just needs an assert here?

Reply:
added in [02c98ee](https://github.com/ar-io/ar-io-network-process/pull/175/commits/02c98ee9f59a74ea8c7f2f6398c13c2b683c96e7)